directory/cli/src/user.py

```diff
@@ -74,7 +74,93 @@ def show(login):
+                        'first': click.prompt('  First name'),
+                        'last': click.prompt('  Surname'),
+                        'email': click.prompt('  Email'),
+            }
+            if click.confirm('Would you like to add a UID to this user?'):
+                params = { **params, 'uid': click.prompt('  UID') }
+
+            group_id = _get_group_id('clusterusers')
+
+            if group_id:
+                params = { **params, 'gidnumber': group_id[0] }
+
+            wrapper = ipa_wrapper_command.create_ipa_wrapper(
+                'user-add',
+                argument_name='login',
+                transform_options_callback=_transform_create_options,
+                handle_result_callback=_handle_create_result,
+            )
+            wrapper(**params)
+
+        @user.command(help='Modify an existing user')
+        def modify():
+            click.echo('Please enter the name of the user you want to modify:')
+            user = click.prompt('  Username')
+            user_find_args = ['--login={}'.format(user)]
+            try:
+                user_data = list_command.find_data(
+                        'user-find',
+                        user_find_args,
+                        all_fields=True
+                )[0]
+            except IpaRunError:
+                # Matches error shown for user show, can extract logic to a method
+                # in the future
+                error = '{}: user not found'.format(user)
+                raise click.ClickException(error)
+
+            click.echo('Modifying user: %s' % user)
+            params = {
+                'login': user,
+                'first': click.prompt(
+                    '  First name',
+                    default=user_data.get('First name', 'empty')[0]
+                ),
+                'last': click.prompt(
+                    '  Surname',
+                    default=user_data.get('Last name', 'empty')[0]
+                ),
+                'email': click.prompt(
+                    '  Email',
+                    default=user_data.get('Email address', 'empty')[0]
+                )
+            }
+
+            wrapper = ipa_wrapper_command.create_ipa_wrapper(
+                'user-mod',
+                argument_name='login',
+                transform_options_callback=_transform_modify_options,
+                handle_result_callback=_handle_modify_result,
+            )
+            wrapper(**params)
+
+    base_ipa_commands = [
+        ipa_wrapper_command.create(
+            'enable',
+            ipa_command='user-enable',
+            argument_name='login',
+            transform_options_callback=_transform_options,
+            help='Enable a user',
+        ),
+        ipa_wrapper_command.create(
+            'disable',
+            ipa_command='user-disable',
+            argument_name='login',
+            transform_options_callback=_transform_options,
+            help='Disable a user',
+        )
```

Comment:
These wrapper commands are duplicated here and below - rather than redefining them and risking them getting out-of-sync, I think it would be better below to define the `advanced_ipa_wrapper_commands` as `advanced_ipa_wrapper_commands = base_ipa_wrapper_commands + [ ... new advanced-only commands]`.